src/main/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -292,7 +294,9 @@ public static LinearIndex createLinearIndex(final File inputFile, final FeatureC
     public static void writeIndex(final Index idx, final File idxFile) throws IOException {
         LittleEndianOutputStream stream = null;
         try {
-            stream = new LittleEndianOutputStream(new BufferedOutputStream(new FileOutputStream(idxFile)));
+            final OutputStream underlyingStream = (idx instanceof TabixIndex) ?
+                    new BlockCompressedOutputStream(idxFile) : new FileOutputStream(idxFile);
```

Comment:
would it be possible for the index to know what type of outputstream it needs and either open it for the factory, or write itself to that stream (given a file). it seems that it's more appropriate to give that decision to the index rather than to the factory.

also: tests please.